core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -886,12 +889,19 @@ static boolean shouldIgnoreOrRejectNewClusterState(ESLogger logger, ClusterState
         }
     }
 
-    private void handleJoinRequest(final DiscoveryNode node, final MembershipAction.JoinCallback callback) {
+    void handleJoinRequest(final DiscoveryNode node, final MembershipAction.JoinCallback callback) {
 
         if (!transportService.addressSupported(node.address().getClass())) {
             // TODO, what should we do now? Maybe inform that node that its crap?
             logger.warn("received a wrong address type from [{}], ignoring...", node);
         } else {
+            // Sanity checks: likely we don't end up here, because serialization is likely to fail.
```

Comment:
maybe my choice of wording here... I'll change it to: `we may not end up here.....`